src/solution.c

```diff
@@ -314,6 +314,50 @@ static void timer_set_period_check(uint32_t timer_peripheral, uint32_t period)
   __asm__("CPSIE i;");
 }
 
+static void simulation_thread()
```

Comment:
maybe pick a different name, e.g. `solution_simulation` or something. `_thread` to me implies this is its own thread but really its running from the solution thread.